plugins/janus_videoroom.c

```diff
@@ -3546,16 +3546,32 @@ static json_t *janus_videoroom_process_synchronous_request(janus_videoroom_sessi
 		GHashTableIter iter;
 		gpointer value;
 		g_hash_table_iter_init(&iter, rooms);
+		gboolean lock_room_list = TRUE;
+		if(admin_key != NULL) {
+			json_t *admin_key_json = json_object_get(root, "admin_key");
+			/* Verify admin_key if it was provided */
+			if(admin_key_json != NULL && strlen(json_string_value(admin_key_json)) > 0) {
```

Comment:
This should also have a check on `json_is_string(admin_key_json)`, as there's no guarantee `admin_key` is a string in the JSON (it's not part of the objects we use for validation in "list", since list requires no arguments).